src/globus_sdk/services/search/client.py

```diff
@@ -14,10 +14,13 @@
 log = logging.getLogger(__name__)
 
 
+@utils.inject_sphinx_params_of(client.BaseClient)
 class SearchClient(client.BaseClient):
```

Comment:
Is there a way to do this without the annotation by looking at the client class's mro instead of annotating?

Reply:
I'm pretty strongly against this being _only_ implicit, but I would be okay with an implicit default. 

Imagining this as a sphinx directive, I'd be okay with
```
.. sdk-autoparams:: BaseClient
.. sdk-autoparams
```
as equivalent. 

The reason I want explicit support is that I can imagine usages where this isn't inheritance based, but sibling methods share params. Especially for mapped/guest collections, which sometimes get slightly different but closely related APIs.